### ISSUE TITLE
Subnetwork API call does not send right Zone data

### ISSUE THREAD

**Issue by hdospitia** [terraform-google-modules/terraform-google-kubernetes-engine#1730]
### TL;DR

When I try to create a cluster with VPC sharing, I pass the NETWORK_PROJECT_ID, NETWORK and SUBNETWORK parameters. However, when launch the apply, the API call sent by the module is **https://www.googleapis.com/compute/v1/projects/my-vpc-shared-project/global/networks/my-shared-subnetwork**, using the global endpoint instead of using the regional, as specified in the variable declaration in cluster.tf:
**subnetwork = "projects/${local.network_project_id}/regions/${local.region}/subnetworks/${var.subnetwork}"**

That leads an error **Error: googleapi: Error 400: Pod secondary range "my-shared-subnetwork" not found in network ...

### Expected behavior

To use the regional endpoint as specified in the cluster.tf definition:

**subnetwork = "projects/${local.network_project_id}/regions/${local.region}/subnetworks/${var.subnetwork}"**

### Observed behavior

Looks at the global endpoint:

https://www.googleapis.com/compute/v1/projects/my-vpc-shared-project/global/networks/my-shared-subnetwork

### Terraform Configuration

```hcl
include "root" {
  path   = find_in_parent_folders()
}

include "common_vars" {
  path    = find_in_parent_folders("common_vars.hcl")
  expose  = true
}

terraform {
    source = "tfr:///terraform-google-modules/kubernetes-engine/google//modules/private-cluster?version=27.0.0"
}

inputs = {
    name                        = "${include.common_vars.locals.prefix_name}-gkecluster-dev"
    project_id                  = "${include.common_vars.locals.gcp_project_id}"
    region                      = "${include.common_vars.locals.gcp_region}"
    network_project_id          = "${include.common_vars.locals.gcp_network_project_id}"
    network                     = "${include.common_vars.locals.gcp_network}"
    subnetwork                  = "${include.common_vars.locals.gcp_subnetwork}"
    ip_range_services           = "XX"
    ip_range_pods               = "XX"
    http_load_balancing         = true
    horizontal_pod_autoscaling  = true
    enable_private_endpoint     = true
    enable_private_nodes        = true
    create_service_account      = false
    service_account             = "XX"
    master_authorized_networks  = [
      {
        cidr_block    = "XX",
        display_name  = "XX"
      }
    ]
    master_ipv4_cidr_block      = "XX"
    node_pools_oauth_scopes = {
    all = [
      "https://www.googleapis.com/auth/logging.write",
      "https://www.googleapis.com/auth/monitoring",
      "https://www.googleapis.com/auth/devstorage.read_only"
    ]
  }
    
    node_pools = [
      {
        name                      = "development-node-pool"
        machine_type              = "e2-medium"
        min_count                 = 1
        max_count                 = 5
        local_ssd_count           = 0
        spot                      = true
        disk_size_gb              = 100
        disk_type                 = "pd-standard"
        image_type                = "COS_CONTAINERD"
        enable_gcfs               = false
        enable_gvnic              = false
        auto_repair               = true
        auto_upgrade              = true
        preemptible               = false
        initial_node_count        = 3
      }
    ]   
}
```

### Terraform Version

```sh
Terraform v1.5.6
```

### Additional information

Using Terragrunt
